test/integration/simple_auth_calls.py

```diff
@@ -1,86 +1,114 @@
-        assert response.status_code == 502
-    except:
-        pass
-    
-def test():
-    prepare_test_env()
-    testregister()
-    testlogin()
-    testverify()
-    tear_down()
-
-if __name__ == "__main__":
-    test()
\ No newline at end of file
+class Resource:
+
+    def __init__(self):
+        print("setup")
+        prepare_test_env()
+        atexit.register(tear_down)
+
+class TestResource:
+
+    r = Resource()
+
+    def test_register(self):
+        print("testing register with")
+        try:
+            response = requests.post("http://localhost/api/register", json={
+                "username": testusername,
+                "password": testpassword,
+                "email": testemail
+            })
+        except Exception as e:
+            print(e)
+        try:
+            print(response.content.decode("utf-8") + " " + str(response.status_code) + " " + str(response.text))
+            print("response.json()[\"username\"] == testusername " + str(response.json()["username"] == testusername) + " " + str(response.json()))
+            assert response.json()["username"] == testusername
+            assert response.status_code == 200
+        except:
+            pass
+
+    def test_login(self):
+        print("testing login with")
+        global voucher
+        try:
+            response = requests.post("http://localhost/api/login", json={
+                "username": testusername,
+                "password": testpassword
+            })
+        except Exception as e:
+            print(e)
+        try:
+            print(response.content.decode("utf-8") + " " + str(response.status_code) + " " + str(response.text))
+            print("\"voucher\" in response.json() " + str("voucher" in response.json()))
+            assert "voucher" in response.json()
+            assert response.status_code == 200
+            voucher = response.json()["voucher"]
+        except:
+            pass
+
+    def test_verify(self):
+        print("testing verify with")
+        global voucher
+        try:
+            response = requests.post("http://localhost/api/verify", json={
+                "voucher": voucher
+            })
+        except Exception as e:
+            print(e)
+        try:
+            print(response.content.decode("utf-8") + " " + str(response.status_code) + " " + str(response.text))
+            assert response.status_code == 200
+            assert response.text == "OK"
+        except:
+            pass
+        try:
+            response = requests.post("http://localhost/api/verify", json={
+                "voucher": "invalidvoucher"
+            })
+        except Exception as e:
+            print(e)
+        try:
+            print(response.content.decode("utf-8") + " " + str(response.status_code) + " " + str(response.text))
+            assert response.status_code == 502
+        except:
+            pass
```

Comment:
Integration tests are used mostly for multiple service communication. Here is only one service and several calls to its RESTful API. I think it could be tested with unit tests by means of Golang infrastructure.